desktop/apps/consignments2/test/reducers.js

```diff
@@ -87,6 +87,130 @@ describe('Reducers', () => {
         })
       })
 
+      describe('#logIn', () => {
+        let store
+        let request
+
+        beforeEach(() => {
+          const middlewares = [ thunk ]
+          const mockStore = configureMockStore(middlewares)
+          const userResponse = {
+            body: {
+              user: {
+                id: 'sarah'
+              }
+            }
+          }
+
+          store = mockStore(initialResponse)
+          request = sinon.stub()
+          request.post = sinon.stub().returns(request)
+          request.set = sinon.stub().returns(request)
+          request.send = sinon.stub().returns(userResponse)
+
+          ActionsRewireApi.__Rewire__('request', request)
+          ActionsRewireApi.__Rewire__('sd', { AP: { loginPagePath: 'https://artsy/login' }, CSRF_TOKEN: 'foo' })
+        })
+
+        afterEach(() => {
+          ActionsRewireApi.__ResetDependency__('request')
+          ActionsRewireApi.__ResetDependency__('sd')
+        })
+
+        it('calls the correct actions', () => {
+          const expectedActions = [
+            {
+              type: 'UPDATE_USER',
+              payload: { user: { id: 'sarah' } }
+            },
+            { type: 'INCREMENT_STEP' },
+            { type: 'CLEAR_ERROR' }
+          ]
+          store.dispatch(actions.logIn({ email: 'sarah@sarah.com', password: '1234' })).then(() => {
```

Comment:
My `await` above was a mistake, but I believe this is one way to test redux actions per https://github.com/reactjs/redux/blob/master/docs/recipes/WritingTests.md#example-1 using `redux-mock-store`.